rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java

```diff
@@ -0,0 +1,135 @@
+import uk.gov.hmcts.ccd.definition.store.repository.AccessTypeRolesRepository;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.AccessTypeRolesEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseTypeEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.JurisdictionEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesField;
+import uk.gov.hmcts.ccd.definition.store.repository.model.OrganisationProfileIds;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesJurisdictionResult;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesJurisdictionResults;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesResult;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesRoleResult;
+
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@Api(value = "/api")
+@RequestMapping(value = "/api")
+public class AccessTypeRolesController {
+
+    private final EntityToResponseDTOMapper entityToResponseDTOMapper;
+    private final AccessTypeRolesRepository accessTypeRolesRepository;
+
+    @Autowired
+    public AccessTypeRolesController(EntityToResponseDTOMapper entityToResponseDTOMapper,
+                                     AccessTypeRolesRepository accessTypeRolesRepository) {
+        this.entityToResponseDTOMapper = entityToResponseDTOMapper;
+        this.accessTypeRolesRepository = accessTypeRolesRepository;
+    }
+
+
+    @PostMapping(value = "/retrieve-access-types",consumes = {"application/json"}, produces = {"application/json"})
+    @ApiOperation(value = "Get all accessTypes for all caseTypes", response = AccessTypeRolesJurisdictionResult.class,
+        responseContainer = "List")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "List of valid access types request"),
+        @ApiResponse(code = 401, message = "Unauthorised request"),
+        @ApiResponse(code = 403, message = "Bad request")
+    })
+    public AccessTypeRolesJurisdictionResults retrieveAccessTypeRoles(
+        @Valid @RequestBody OrganisationProfileIds organisationProfileIds) {
+        List<AccessTypeRolesField> accessTypeRoles;
+
+        if (organisationProfileIds != null
+            && organisationProfileIds.getOrganisationProfileIds() != null
+            && !organisationProfileIds.getOrganisationProfileIds().isEmpty()) {
+            //Get only access_types with caseTypeId's by OrganisationProfileIds
+            List<AccessTypeRolesEntity> accessTypeRolesEntities = accessTypeRolesRepository
+                .findByOrganisationProfileIds(organisationProfileIds.getOrganisationProfileIds());
+            accessTypeRoles = accessTypeRolesEntities.stream().map(entityToResponseDTOMapper::map)
+                .collect(Collectors.toList());
+        } else {
+            // get all access Types  with caseTypeId's
```

Comment:
do we need this comment the method findAllWithCaseTypeIds is descriptive enough I think?